### DISCUSSION TITLE
SQL Server - DDL generation for primary key constraint is inadequate

### DISCUSSION THREAD

**Discussion by btgreen001** [community/community#111230] [Programming Help]
### Select Topic Area

Bug

### Feature Area

Issues

### Body

When generating DDL for a table.  The PK constraint is scripted inadequately.  Minimally the constraint lacks the index type (CLUSTERED vs NONCLUSTERED).  In doing so this will results in errors when utilizing the script to alter/create the table when a different index also exists that is CLUSTERED.  When not indicated, SQL Server assumes that the PK is clustered (based upon the lack of designation) and reports that two CLUSTERED indexes are not permitted.  Example script below where TESTCOL1 PK constraint is NONCLUSTERED with a different CLUSTERED index but scripted without PK index type designation.  

Error - SQL Error [1902] [S0003]: Cannot create more than one clustered index on table 'AccessControl.TEST_TABLE'. Drop the existing clustered index 'PK__TEST_TABLE__AC_TESTCOL1' before creating another.

**DDL Script missing PRIMARY KEY being NONCLUSTERED**
CREATE TABLE AccessControl.TEST_TABLE (
	TESTCOL1 int IDENTITY(1,1) NOT NULL,
	TESTCOL2 int NOT NULL,
	CONSTRAINT PK__TEST_TABLE__AC_TESTCOL1 PRIMARY KEY (TESTCOL1)
);
CREATE UNIQUE CLUSTERED INDEX CUIX_AccessControl__TEST_TABLE__TESTCOL2 ON AccessControl.TEST_TABLE (TESTCOL2);

Minimally corrected, the type "NONCLUSTERED" is added....

CREATE TABLE AccessControl.TEST_TABLE (
	TESTCOL1 int IDENTITY(1,1) NOT NULL,
	TESTCOL2 int NOT NULL,
	CONSTRAINT PK__PERT_USE__AC PRIMARY KEY **NONCLUSTERED** (TESTCOL1)
);
CREATE UNIQUE CLUSTERED INDEX CUIX_AccessControl__TEST_TABLE__TESTCOL2 ON AccessControl.TEST_TABLE (TESTCOL2);

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @btgreen001 ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Projects and Issues category is solely related to conversations around the GitHub products Issues and Projects. 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
